### PR TITLE
Debug face from contours

### PR DESCRIPTION
**Pull request by GabrielJMS**
* **Please check if the PR fulfills these requirements**
- [x] The commit messages follows our guidelines (explicit, in English)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [x] Bug fix:
- [ ] New features:
- [ ] Documentation
- [ ] Optimization

* **Other information**:

### CODE REVIEW

File: volmdlr/faces.py [hunk 1]
```diff
@@ -2606,18 +2632,50 @@ def bsplinecurve3d_to_2d(self, bspline_curve3d):
...
+
+        # if theta3 < theta1 < theta2 and theta_discontinuity:
+        #     points = [p - volmdlr.Point2D(volmdlr.TWO_PI, 0) if p.x > 0 else p for p in points]
+        # elif theta3 > theta1 > theta2 and theta_discontinuity:
+        #     points = [p + volmdlr.Point2D(volmdlr.TWO_PI, 0) if p.x < 0 else p for p in points]
+
+        # if phi3 < phi1 < phi2 and phi_discontinuity:
+        #     points = [p - volmdlr.Point2D(volmdlr.TWO_PI, 0) if p.y > 0 else p for p in points]
+        # elif phi3 > phi1 > phi2 and phi_discontinuity:
+        #     points = [p + volmdlr.Point2D(volmdlr.TWO_PI, 0) if p.y < 0 else p for p in points]
```
Comment (WirajanDASILVA) [inline volmdlr/faces.py]: clean?

File: volmdlr/faces.py [hunk 2]
```diff
@@ -1824,16 +1835,91 @@ def bsplinecurve3d_to_2d(self, bspline_curve3d):
...
+
+        # Verify if theta1 or theta2 point should be -pi because atan2() -> ]-pi, pi]
+        if abs(theta1) == math.pi or abs(theta1) == 0.5 * math.pi:
+            theta1 = vm_parametric.repair_start_end_angle_periodicity(theta1, theta3)
+        if abs(theta2) == math.pi or abs(theta2) == 0.5 * math.pi:
+            theta4, _ = points[-2]
+            # make sure that the reference angle is not undefined
+            if abs(theta4) == math.pi:
+                theta4, _ = points[-3]
+            theta2 = vm_parametric.repair_start_end_angle_periodicity(theta2, theta4)
```
Comment (WirajanDASILVA) [inline volmdlr/faces.py:1884]: should all the researches of different thetas be replaced in another method maybe?

File: tests/faces/objects_periodical_surface/contour3d__openned_one_contour_1.json [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        1.0
+      ],
+      "weights": null,
+      "periodic": true
+    }
+  ],
+  "basis_primitives": [],
+  "color": null,
+  "alpha": 1
+}
```
Comment (GabrielJMS) [inline tests/faces/objects_periodical_surface/contour3d__openned_one_contour_1.json:154]:
```suggestion
}

```

File: volmdlr/faces.py [hunk 4]
```diff
@@ -2898,6 +2941,39 @@ def linesegment2d_to_3d(self, linesegment2d):
...
+        for primitive3d in contour3d.primitives:
+            method_name = f'{primitive3d.__class__.__name__.lower()}_to_2d'
+            if hasattr(self, method_name):
+                primitives = getattr(self, method_name)(primitive3d)
+
+                if primitives is None:
+                    continue
+                primitives2d.extend(primitives)
+            else:
+                raise NotImplementedError(f'Class {self.__class__.__name__} does not implement {method_name}')
```
Comment (WirajanDASILVA) [inline volmdlr/faces.py:2964]: should the search for the primitives in the parametric domain be transformed into a separate method?

File: volmdlr/faces.py [hunk 5]
```diff
@@ -4436,36 +4526,38 @@ def bsplinecurve3d_to_2d(self, bspline_curve3d):
...
+                                if should_be_umax:
+                                    points[0] = volmdlr.Point2D(points[0].x, u_max)
+                                else:
+                                    points[-1] = volmdlr.Point2D(points[-1].x, u_max)
+                            elif math.isclose(points[0].y, u_max, abs_tol=1e-6):
+                                should_be_umin = (u_max - points[1].y) > (points[1].y - u_min)
+                                if should_be_umin:
+                                    points[0] = volmdlr.Point2D(points[0].x, u_min)
+                                else:
+                                    points[-1] = volmdlr.Point2D(points[-1].x, u_min)
```
Comment (WirajanDASILVA) [inline volmdlr/faces.py:4560]: keep change from last merge

File: volmdlr/faces.py [hunk 6]
```diff
@@ -9428,7 +9532,10 @@ def point_in_shell_face(self, point: volmdlr.Point3D):
 
     def triangulation(self):
         meshes = []
-        for face in self.faces:
+        for i, face in enumerate(self.faces):
+            # print(i)
+            if i == 1050:
+                print(True)
```
Comment (WirajanDASILVA) [inline volmdlr/faces.py:9538]: clean?

**Review by WirajanDASILVA — CHANGES_REQUESTED** [review]

File: volmdlr/wires.py [hunk 7]
```diff
@@ -1667,8 +1667,8 @@ def _get_edge_polygon(self):
                     points.append(edge.start)
             else:
                 points.append(edge.start)
-        return ClosedPolygon2D(points)
-
+        closedpolygon = ClosedPolygon2D(points)
+        return closedpolygon
```
Comment (WirajanDASILVA) [inline volmdlr/wires.py:1671]:
```suggestion
        return closedpolygon

```

**Review by WirajanDASILVA — CHANGES_REQUESTED** [review]

**Review by WirajanDASILVA — APPROVED** [review]

**Pull request merged**